rfc/0036-client-cert-auth.md

```diff
@@ -0,0 +1,130 @@
+
+In all three cases, the SDKs must check for that specific error and IF x509 is enabled through the configuration, raise a `CertAuthNotSupportedException` or a similar error code to make it clear to the user whats going on (since the individual errors are ambiguous).
+
+### KV Engine And X.509
+As soon as the certificate with credentials is passed to KV engine over 11210, subsequent SASL authentication mechanisms are not supported anymore. This means that for example a LIST_MECHS returns with a 0x83 NOT_SUPPORTED, since the auth has already been performed.
+
+As a result, as soon as client certificate auth is enabled, all SASL steps need to be skipped and not performed as part of the bootstrap negotiations. Note that SELECT_BUCKET is not considered to be part of SASL auth and as a result still needs to be performed (otherwise since just a username is given, a bucket to use still needs to be negotiated).
+
+### Client Bootstrap
+
+- When everything is set up properly but then openBucket gets called with a password, a `AuthenticationException` or similar needs to be thrown indicating that this combination doesn’t work. The exception/error is similar to the one where RBAC authentication is used and a password is provided during open bucket.
+- In SDKs where possible illegal combination of Authenticator and SSL credential input (key, keystore, truststore…) should be detected and raised. This is optional and not specified in the RFC because of platform specific details.
+
+Specifically the following cases should be handled (and to be raised with a AuthenticationException compatible error from the language table below):
+
+A CertAuthenticator is used AND bucket credentials are specified on openBucket().
+
+- If the environment/connection string is configured to use certificate authentication but no CertAuthenticator OR a different Authenticator is specified
+- If a `CertAuthenticator` is used but the environment/connection string is not configured with the proper configuration.
+
+## Certificate and Credentials
+It is also possible that a certificate is provided which does not contain credentials. The SDK needs to properly react to different circumstances as the following table shows:
+
+|                   | Certificate Without Creds                                                                                | Certificate Without Creds                                                                          |
+|-------------------|----------------------------------------------------------------------------------------------------------|----------------------------------------------------------------------------------------------------|
+| certAuth Enabled  | - Expected - uses credentials from cert - Errors might still happen to unsupportive services (see above) | - SASL is skippedno http auth header sent,  - leading to auth errors - Doesn’t work                |
```

Comment:
`skippedno` should have space

Also could you use single quote instead of `’` ([RIGHT SINGLE QUOTATION MARK](https://www.fileformat.info/info/unicode/char/2019/index.htm))? The same for `“` ([LEFT DOUBLE QUOTATION MARK](https://www.fileformat.info/info/unicode/char/201c/index.htm)) and `”` ([RIGHT DOUBLE QUOTATION MARK](https://www.fileformat.info/info/unicode/char/201d/index.htm))